api/envoy/extensions/filters/http/ext_proc/v3alpha/processing_mode.proto

```diff
@@ -0,0 +1,80 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.http.ext_proc.v3alpha;
+
+import "udpa/annotations/status.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.http.ext_proc.v3alpha";
+option java_outer_classname = "ProcessingModeProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).work_in_progress = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: External Processing Filter]
+// External Processing Filter Processing Mode
+// [#extension: envoy.filters.http.ext_proc]
+
+// This configuration describes how an external processing filter processes the HTTP
+// request or response, including how requests and responses are
+// sent to the remote server.
+
+// [#next-free-field: 9]
+message ProcessingMode {
+  enum SyncMode {
+    // The default SyncMode depends on which part of the message is being set.
+    SYNC_DEFAULT = 0;
+
+    // Send the header and wait for a response before continuing with
+    // the filter chain.
+    SYNC = 1;
+
+    // Send the header but to not expect a response, continuing immediately
+    // with the filter chain.
+    ASYNC = 2;
+
+    // Do not send the header.
+    SKIP = 3;
+  }
+
+  enum BodyMode {
+    // The default SyncMode depends on which part of the message is being set.
+    BODY_DEFAULT = 0;
+
+    // Send each chunk of the body separately to the server.
+    CHUNKED = 1;
+
+    // Buffer the message body in memory and send the entire body in one
+    // chunk. If the body exceeds the configured buffer limit, then the
+    // downstream system will receive an error.
+    BUFFERED = 2;
+
+    // Buffer the message body in memory and send the entire body in one
+    // chunk. If the body exceeds the configured buffer limit, then the body contents
+    // up to the buffer limit will be sent.
+    BUFFERED_PARTIAL = 3;
+  }
+
+  // How to handle the request header. Default is "SYNC".
+  ProcessingMode request_header_mode = 1;
```

Comment:
Is it legal to have async headers and sync body?  Mixing and matching seems confusing and dangerous.

Alternately, would it make sense to have  one blocking/non-blocking boolean for request/response  and "send or don't send" booleans for different parts of the request/response?

Reply:
+1, mixing and matching sync vs async on the same service feels confusing and error-prone.

Are we sure we even need this flexibility from the get-go? i.e. can the initial implementation be "always wait for a response before continuing with the filter chain" and we can add the fire-and-forget functionality later? I'm having trouble imagining a use case for sending the headers to an external service but not caring what the response is anyway.